### ISSUE TITLE
Hidden properties and virtual properties not works with Embeddables

### ISSUE THREAD

**Issue by ah-nvision** [mikro-orm/mikro-orm#3429]
**Describe the bug**
Serializing embeddable with `@Property({ hidden: true })` and virtual properties not works like entity

``` typescript
@Embeddable()
class Address {
  @Property({ hidden: true })
  addressLine1: string;

  @Property({ hidden: true })
  addressLine2: string;

  @Property({ persist: false })
  get fullAddress() {
    return [this.addressLine1, this.addressLine2].join(' ');
  }

  @Property()
  city: string;

  @Property()
  country: string;
}

@Entity()
export class Organization {
  @PrimaryKey()
  id: string;

  @Embedded(() => Address, { object: true })
  address: Address;
}
```

Example output for above schema
```json
{
  "id": "...id...",
  "address": {
    "addressLine1": "Line 1",
    "addressLine2": "Line 2",
    "city": "City",
    "country": "Country"
  }
}
```

**Expected behavior**
Expected output to be
```json
{
  "id": "...id...",
  "address": {
    "fullAddress": "Line 1 Line 2",
    "city": "City",
    "country": "Country"
  }
}
```

CodeSandbox: [https://codesandbox.io/s/sweet-chaum-htf28x](https://codesandbox.io/s/sweet-chaum-htf28x)

**Versions**

| Dependency | Version |
| - | - |
| node | 16.14.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.3.1 |
| better-sqlite | 5.3.1 |

**Comment by B4nan** [comment]
Interesting, thanks for the repro. This is probably because they are not entities (they won't pass the `Utils.isEntity()` check). 

But I dont see any reason why we shouldn't support this.

sidenote: I should prepare some simpe repro codesandbox and add it to the issue template, this is very nice. Could be in a "single file repro" like the existing issue tests, with with all the setup around so people dont need to clone the repo at all.

**Comment by JasonRammoray** [comment]
I've faced that issue too.

As a **temporary** workaround, one might add a custom `toJSON` serializer to an entity class :
```typescript
@Entity(<metadataHere>)
class SomeEntity {
  @Property({hidden: true})
  // the actual type doesn't matter for the purpose of this demo
  someProp: string;

  toJSON(...args: string[]) {
    return wrap<SomeEntity>(this).toObject(args);
  }
}
```

@B4nan, look forward to seeing this fixed.
Thank you.

**Comment by ah-nvision** [comment]
Hi @B4nan, this issue seems not fixed, can reproduce in v5.4.0 and v5.4.1

**Comment by B4nan** [comment]
Hmm right, the test I used was producing false positives. Weird that it didnt help, lets reopen.

Here is afailing stackblitz, I had some issues with codesandbox. I would appreciate simple repros like this one, as a script, no web apps when they are unrelated to the issue.

https://stackblitz.com/edit/mikro-orm-repro-q8nadr?file=test.ts
